src/ContentSecurityPolicy/DirectiveSet.php

```diff
@@ -140,6 +140,9 @@ public function buildHeaderValue(Request $request, ?array $signatures = null): s
                 // this is the recommended way to deal with this.
                 if (false === strpos($value, '\'unsafe-inline\'') && $this->level1Fallback) {
                     $policy[] = $name.' '.$value.' \'unsafe-inline\' '.$signatures[$name];
+                // Do not add any signatures if 'unsafe-inline' is allowed anyway
+                } elseif (false !== strpos($value, '\'unsafe-inline\'')) {
+                    $policy[] = $name.' '.$value;
```

Comment:
This is risky IMO, because if you use signatures then unsafe-inline will be ignored by modern browsers supporting CSP2. It's merely there as a fallback.

Now with your change suddenly having unsafe-inline in an existing config would silently break/remove signatures.

I think the more appropriate way is to disable the hash functionality if you are not interested in outputting hashes.